cmd/lp-cc/main.go

```diff
@@ -0,0 +1,136 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"strconv"
+	"strings"
+
+	_ "net/http/pprof"
+
+	"github.com/ScottSallinen/lollipop/enforce"
+	"github.com/ScottSallinen/lollipop/framework"
+	"github.com/ScottSallinen/lollipop/graph"
+)
+
+func info(args ...any) {
+	log.Println("[CC]\t", fmt.Sprint(args...))
+}
+
+func EdgeParser(lineText string) graph.RawEdge[EdgeProperty] {
+	stringFields := strings.Fields(lineText)
+
+	sflen := len(stringFields)
+	enforce.ENFORCE(sflen == 2 || sflen == 3)
+
+	src, _ := strconv.Atoi(stringFields[0])
+	dst, _ := strconv.Atoi(stringFields[1])
+
+	return graph.RawEdge[EdgeProperty]{SrcRaw: uint32(src), DstRaw: uint32(dst)}
+}
+
+// OnCheckCorrectness: Performs some sanity checks for correctness.
+func OnCheckCorrectness(g *graph.Graph[VertexProperty, EdgeProperty, MessageValue]) error {
+	// Denote vertices that claim unvisted, and ensure out edges are at least as good as we could provide
```

Comment:
Check comments. Please remove old ones for copy past from SSSP, and make new ones to describe the new behaviour.